src/libopensc/opensc.h

```diff
@@ -940,9 +946,11 @@ int sc_cancel(sc_context_t *ctx);
 /**
  * Tries acquire the reader lock.
  * @param  card  The card to lock
+ * @param  resetok If nonzero means to continue even if the card was reset
+ *	as long as it wasn't reset between individual operations
  * @retval SC_SUCCESS on success
  */
-int sc_lock(struct sc_card *card);
+int sc_lock(struct sc_card *card, int resetok);
```

Comment:
> This is currently unused but it can be used in a future code...

I don't think we should add unused code.

> ... supporting cards with multiple applets...

@dengert introduced 84a69ce2ba100be8121218a61a5d549ea3cbdbfb, which can be used to bring the card back into a usable state. 84a69ce2ba100be8121218a61a5d549ea3cbdbfb, by itself, [is debatable](https://github.com/OpenSC/OpenSC/pull/842#issuecomment-239583048). I don't think we should add an other half baked solution to the lowlevel functionality in libopensc. The more I think about it, the more I tend to `card->ops->card_reader_lock_obtained(card, was_reset);` should be called instead if a reset was detected (either during an actual lock or a transmit). The card driver (in your case OpenPGP) should then re-select the applet if needed.

In the minidriver, however, your state tracking may still make sense.